From ec1a8e737e6b7d495c67f3fd0ac7546a7fad80d3 Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jiri@resnulli.us>
Date: Thu, 3 Jan 2013 22:48:57 +0000
Subject: [PATCH 431/858] qlcnic: guard __vlan_find_dev_deep() by rcu_read_lock

rcu_read_lock was missing here

Signed-off-by: Jiri Pirko <jiri@resnulli.us>
Acked-by: Sony Chacko <sony.chacko@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 43c00a75d205ffee165c40f37edb1d2173dea7ea)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index 5331fda..fb7ac8e 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -3185,12 +3185,14 @@ void qlcnic_restore_indev_addr(struct net_device *netdev, unsigned long event)
 
  qlcnic_config_indev_addr(adapter, netdev, event);
 
+ rcu_read_lock();
  for_each_set_bit(vid, adapter->vlans, VLAN_N_VID) {
   dev = __vlan_find_dev_deep(netdev, vid);
   if (!dev)
    continue;
   qlcnic_config_indev_addr(adapter, dev, event);
  }
+ rcu_read_unlock();
 }
 
 static int qlcnic_netdev_event(struct notifier_block *this,
-- 
1.8.3.1